scripts/game_structure/windows.py

```diff
@@ -621,6 +621,63 @@ def process_event(self, event):
                 game.all_screens['profile screen'].screen_switches()
                 self.kill()
 
+class ExileCat(UIWindow):
+    """This window allows the user to specify the reason for exile"""
+    def __init__(self, cat):
+        super().__init__(scale(pygame.Rect((600, 400), (900, 400))),
+                         window_display_title='Exile Cat',
+                         object_id='#change_cat_gender_window',
+                         resizable=False)
+        self.history = History()
+        game.switches['window_open'] = True
+        self.the_cat = cat
+        self.back_button = UIImageButton(
+            scale(pygame.Rect((840, 10), (44, 44))),
+            "",
+            object_id="#exit_window_button",
+            container=self
+        )
+        self.heading = pygame_gui.elements.UITextBox(f"<b>-- How was this cat exiled? --</b>",
+                                                     scale(pygame.Rect((20, 20), (860, 150))),
+                                                     object_id="#text_box_30_horizcenter_spacing_95",
+                                                     manager=MANAGER,
+                                                     container=self)
+
+        self.initial = 'It was the decision of the clan to exile this cat.'
+        self.prompt = None
+
+        self.exile_entry_box = pygame_gui.elements.UITextEntryBox(scale(pygame.Rect((50, 110), (800, 150))),
+                                                                      initial_text=self.initial,
+                                                                      object_id="text_entry_line",
+                                                                      manager=MANAGER,
+                                                                      container=self)
+
+        self.done_button = UIImageButton(scale(pygame.Rect((373, 305), (154, 60))), "",
+                                             object_id="#done_button",
+                                             manager=MANAGER,
+                                             container=self)
+
+        self.set_blocking(True)
+
+    def process_event(self, event):
+        super().process_event(event)
+
+        if event.type == pygame_gui.UI_BUTTON_START_PRESS:
+            if event.ui_element == self.done_button:
+                exile_reason = sub(r"[^A-Za-z0-9<->/.()*'&#!?,| ]+", "", self.exile_entry_box.get_text())
+                #self.the_cat.exile_reason = exile_reason
+                self.the_cat.exile()
+                self.history.add_death_or_scars(self.the_cat, text=exile_reason,exile=True)
```

Comment:
It it best to create a new function to add exile reason, since "add_death_and_scars" is already used for a lot of purposes. That particular function also needs to be re-done, since it causing a few bugs right now.